src/members/Members.jsx

```diff
@@ -1,21 +1,93 @@
-import React from "react";
+import React, { useState, useContext, useEffect } from "react";
+import { Table, Button } from "react-bootstrap";
+import { Route } from "react-router-dom"
+import { AuthContext } from "../AuthContext";
+import { BASE_API } from "../config";
+import { SERVICE_UNAVAILABLE_ERROR } from "../messages";
+import MemberProfile from "./MemberProfile";
 
 export default function Members() {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [members, setMembers] = useState([]);
+  const { access_token } = useContext(AuthContext);
+
+  const requestMembersList = {
+    method: "GET",
+    headers: {
+      "Authorization": `Bearer ${access_token}`,
+      "Accept": "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  useEffect(() => {
+    fetch(`${BASE_API}/users`, requestMembersList)
+      .then(async response => {
+        const data = await response.json();
+        if (response.ok) {
+          return setMembers(data);
+        }
+        setErrorMessage(data.message);
+      })
+      .catch(error =>
+        setErrorMessage(SERVICE_UNAVAILABLE_ERROR)
+      )
+
+  }, [] );
+ 
+  const handleClick = ({member}) => {
     return (
-        <div className="container-fluid" id="members">
-          <div className="top">
-            <h1>
-              This is the Public page for Members list
-            </h1>
-          </div>
-          <div className="middle">
-            <h2>The site is currently under construction...</h2>
-          </div>
-          <div className="bottom">
-            <h3>
-                Thank you for your patience and understanding.
-            </h3>
+      <Route render={() => {
+          return (
+            <div>
+              <MemberProfile member={member} />
```

Comment:
No need to use Route here again. You have already defined a route that is /member-profile. What you want to do on click is simply redirect to member-profile. That route will render the MemberProfile component. 

Find an example of how, useHistory hook is used here - https://css-tricks.com/the-hooks-of-react-router/

Reply:
I've pushed my latest commit, but the redirecting still doesn't work. I hope we can discuss this on your office hour tonight.

Reply:
@meenakshi-dhanani .Looking through Tyler McGinnis, useHistory is just another option (imperative navigation) on React Router v5. The one that I tried to use here is <Redirect ... /> which is the declarative way that he is in favor of using for navigating.